cudapoa/src/cudapoa_nw_banded.cuh

```diff
@@ -556,6 +556,124 @@ __device__ __forceinline__
     return aligned_nodes;
 }
 
+// global kernel used in testing, hence uses int16_t for SizeT and ScoreT,
+// may need to change if test inputs change to long reads
+template <typename SizeT, bool ADAPTIVE>
+__global__ void runNeedlemanWunschBandedKernel(uint8_t* nodes,
+                                               SizeT* graph,
+                                               SizeT* node_id_to_pos,
+                                               int32_t graph_count,
+                                               uint16_t* incoming_edge_count,
+                                               SizeT* incoming_edges,
+                                               uint16_t* outgoing_edge_count,
+                                               uint8_t* read,
+                                               int32_t read_length,
+                                               int16_t* scores,
+                                               int32_t scores_width,
+                                               int32_t max_nodes_per_graph,
+                                               SizeT* alignment_graph,
+                                               SizeT* alignment_read,
+                                               int32_t band_width,
+                                               int32_t gap_score,
+                                               int32_t mismatch_score,
+                                               int32_t match_score,
+                                               SizeT* aligned_nodes)
+{
+    float banded_buffer_size = static_cast<float>(max_nodes_per_graph) * static_cast<float>(scores_width);
```

Comment:
Will precision round-off be an issue for a corner case? (I realize that it might be pretty rare)

Reply:
`banded_buffer_size` is to determine maximum capacity for scores buffer (or in some similar cases, traceback buffer). This maximum is based on band-width and maximum graph size, which the latter is selected heuristically, e.g. x3 of max sequence size. So overall what `banded_buffer_size` implies can certainly tolerate some error. The reason behind strange-looking cast to float is minimizing register count, somehow compiler did not like int64_t and switching to float would save around 4-5 registers if I remember correctly.

Reply:
Ok!